nucleus/admin/monitor/src/main/java/fish/payara/admin/monitor/cli/ResetMonitoringLevels.java

```diff
@@ -0,0 +1,136 @@
+    String target;
+
+    @Param(name = "verbose", optional = true, shortName = "v")
+    private boolean verbose;
+
+    @Inject
+    private Target targetUtil;
+
+    @Inject
+    private Logger logger;
+
+    private MonitoringService monitoringService;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        ActionReport actionReport = context.getActionReport();
+
+        List<String> validModuleList = Arrays.asList(Constants.validModuleNames);
+        Config config = targetUtil.getConfig(target);
+        if (config != null) {
+            monitoringService = config.getMonitoringService();
+        } else {
+            actionReport.setMessage("Cound not find target: " + target);
+            actionReport.setActionExitCode(ActionReport.ExitCode.FAILURE);
+            return;
+        }
+
+        Map<String, String> enabledModules = getPreviouslyEnabledModules(validModuleList);
+
+        //Setting all modules to "OFF"
+        for (String module : validModuleList) {
+            try {
+                ConfigSupport.apply((final MonitoringService monitoringServiceProxy) -> {
+                    monitoringServiceProxy.setMonitoringLevel(module, "OFF");
+                    return monitoringServiceProxy;
+                }, monitoringService);
+            } catch (TransactionFailure ex) {
+                logger.log(Level.WARNING, "Failed to execute the command set-monitoring-level, while setting modules to 'OFF': {0}", ex.getCause().getMessage());
+                actionReport.setMessage(ex.getCause().getMessage());
+                actionReport.setActionExitCode(ActionReport.ExitCode.FAILURE);
+            }
+        }
+
+        //Setting modules back to what they were previously
+        for (String module : enabledModules.keySet()) {
+            String moduleLevel = enabledModules.get(module);
+            try {
+                ConfigSupport.apply((final MonitoringService monitoringServiceProxy) -> {
+                    monitoringServiceProxy.setMonitoringLevel(module, moduleLevel);
+                    return monitoringServiceProxy;
+                }, monitoringService);
+            } catch (TransactionFailure ex) {
+                logger.log(Level.WARNING, "Failed to execute the command set-monitoring-level, while setting modules to previous values: {0}", ex.getCause().getMessage());
+                actionReport.setMessage(ex.getCause().getMessage());
+                actionReport.setActionExitCode(ActionReport.ExitCode.FAILURE);
+            }
+        }
+
+        if (verbose) {
+            actionReport.setMessage(getFormattedColumns(enabledModules).toString());
+        } else {
+            actionReport.setMessage("Reset " + enabledModules.size() + " modules");
+        }
+    }
+
+    private Map<String, String> getPreviouslyEnabledModules(List<String> validModules) {
+        Map<String, String> enabledModules = new HashMap<>();
+        if (!validModules.isEmpty()) {
```

Comment:
Usually such a check before a for loop isn't needed unless you have a suspicion that creating an iterator is a surprisingly expensive operation for the list while `isEmpty` wouldn't be.